src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3268,6 +3279,25 @@ def set_up_agentpool_profile_ssh_access(self, mc: ManagedCluster) -> ManagedClus
                 agent_pool_profile.security_profile.ssh_access = ssh_access
         return mc
 
+    def set_up_bootstrap_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        self._ensure_mc(mc)
+
+        bootstrap_artifact_source = self.context.get_bootstrap_artifact_source()
+        bootstrap_container_registry_resource_id = self.context.get_bootstrap_container_registry_resource_id()
+        if bootstrap_artifact_source is not None:
+            if bootstrap_artifact_source != CONST_ARTIFACT_SOURCE_CACHE and bootstrap_container_registry_resource_id:
+                raise MutuallyExclusiveArgumentError(
+                    "Cannot specify --bootstrap-container-registry-resource-id when "
+                    "--bootstrap-artifact-source is not Cache."
+                )
+            if mc.bootstrap_profile is None:
+                mc.bootstrap_profile = self.models.ManagedClusterBootstrapProfile()  # pylint: disable=no-member
+            mc.bootstrap_profile.artifact_source = bootstrap_artifact_source
+            if bootstrap_artifact_source == CONST_ARTIFACT_SOURCE_CACHE:
```

Comment:
>CONST_ARTIFACT_SOURCE_CACHE [](http://example.com/codeflow?start=44&length=27)

no need check? already checked bootstrap_container_registry_resource_id must be None when not cache